File: Week4/HW2/Pages/homepage.py
import time

class HomePage():
    def __init__(self, driver):
        self.driver = driver

        self.scroll_to_top = self.driver.execute_script("scroll(0, 0);")
        self.basket_button = self.driver.find_element_by_xpath("//span[@class='small--hide']")


    def goto_homepage(self):
        self.driver.get("https://thebouqet.com/")

    def add_item_to_basket(self, index):
        self.driver.find_element_by_xpath(f'//*[@id="shopify-section-featured-products"]/div/div/div[{index}]').click()
        self.driver.find_element_by_xpath("//span[@id='AddToCartText']").click()


    def get_item_number_at_basket(self):
        return int(self.driver.find_element_by_xpath('//*[@id="CartCount"]').get_attribute("textContent"))

File: Week4/HW1/PageObject/HomePage.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Threading;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Firefox;
using OpenQA.Selenium.Remote;
using OpenQA.Selenium.Support.UI;
using OpenQA.Selenium.Interactions;
using NUnit.Framework;
using SeleniumExtras.PageObjects;
using dotNET_Test.TestClasses;

namespace dotNET_Test.PageObjects {

  public class HomePage : BasePage {
      public HomePage (IWebDriver driver) {
        HomePage.driver = driver;
        PageFactory.InitElements(driver,this);
      }

    public void addToBasket(){
        driver.FindElement(By.XPath("//*[@id='shopify-section-featured-products']/div/div/div[1]")).Click();
        driver.FindElement(By.XPath("//span[@id='AddToCartText']")).Click();
    }

    public void goToBasket(){
      driver.FindElement(By.XPath("//span[@class='small--hide']")).Click();
    }

  }
}